Added TD Agent bit support and some service options

Angelo-Verona opened the pull request:
- Added support for TD Agent Bit
- Added more config service options

Commit:
- Added support for TD Agent Bit
- Added more config service options

aabouzaid commented on fluent-bit/map.jinja:
```diff
@@ -20,7 +20,7 @@ that differ from whats in defaults.yaml
     'systemd': {
       "service": {
-        "conf": '/lib/systemd/system/fluent-bit.service',
+        "conf": '/etc/init/fluent-bit.conf',
```
Why this change? This is not `systemd` conf.

Angelo-Verona replied:
Hm, this looks like as mistake on my side, I need to check that. Thanks for catch

aabouzaid commented on fluent-bit/defaults.yaml:
```diff
@@ -1,10 +1,16 @@
 # vim: ft=yaml
 fluent_bit:
-  pkg: fluentbit
+  pkg: fluent-bit
```
The official distributed package is called `td-agent-bit`, should we use it as a default name?

Angelo-Verona replied:
Fluent-bit it's open source, and to-agent-bit is its commercial version. I lean to opensource more.
If this formula will be able to install both of them, it probably doesn't matter.
Maybe I would opt for just short variables like "bit" and allow merge from both top pillars tree like fluent_bit or td_agent_bit. I have in general issues if variables in Salt using dashes "-" sometimes it's working sometimes don't.

aabouzaid commented on fluent-bit/config.sls:
```diff
@@ -3,9 +3,9 @@
 {% from "fluent-bit/map.jinja" import fluent_bit with context %}
 
-fluent-bit-config:
+{{ fluent_bit.pkg }}-config:
```
It's better to use `{{ sls }}` so it uses the directory name whatever is it.

Angelo-Verona replied:
You are right, but package name is already sanitized, and directory can have any characters which can break things. The package name will give you still the output what you are installing for "version" right now even if future fork of anything-bit will be released and you will change pkg package name.

Commit:
Fixed systemd service
- Refactored fluent_bit to "bit" and confusing variables
  pillar still perserve fluent_bit to not break compatibility
- Better config formating
- Added Description detection for systemd unit file

Angelo-Verona commented:
I have new pull request with additional changes and fixes
